File: src/lib/api/schedule.ts
import { invoke } from '@tauri-apps/api/core';
import type { MonthScheduleView, ScheduleView } from '../types';

export const getSchedule = (eventId: string) => invoke<ScheduleView>('get_schedule', { eventId });
export const generateSchedule = (eventId: string) => invoke<ScheduleView>('generate_schedule', { eventId });
export const clearSchedule = (eventId: string) => invoke<void>('clear_schedule', { eventId });

export const getMonthSchedule = (month: string) => invoke<MonthScheduleView>('get_month_schedule', { month });
export const generateMonthSchedule = (month: string) => invoke<MonthScheduleView>('generate_month_schedule', { month });
export const clearMonthSchedule = (month: string) => invoke<void>('clear_month_schedule', { month });

File: src-tauri/src/commands/schedule.rs
use crate::{errors::AppError, models::schedule::{MonthScheduleView, ScheduleView}, services::schedule_service, AppState};
use tauri::State;

#[tauri::command] pub async fn get_schedule(state: State<'_, AppState>, event_id: String) -> Result<ScheduleView, AppError> { schedule_service::get_schedule(&state.db, &event_id).await }
#[tauri::command] pub async fn generate_schedule(state: State<'_, AppState>, event_id: String) -> Result<ScheduleView, AppError> { schedule_service::generate_schedule(&state.db, &event_id).await }
#[tauri::command] pub async fn clear_schedule(state: State<'_, AppState>, event_id: String) -> Result<(), AppError> { schedule_service::clear_schedule(&state.db, &event_id).await }

#[tauri::command] pub async fn get_month_schedule(state: State<'_, AppState>, month: String) -> Result<MonthScheduleView, AppError> { schedule_service::get_month_schedule(&state.db, &month).await }
#[tauri::command] pub async fn generate_month_schedule(state: State<'_, AppState>, month: String) -> Result<MonthScheduleView, AppError> { schedule_service::generate_month_schedule(&state.db, &month).await }
#[tauri::command] pub async fn clear_month_schedule(state: State<'_, AppState>, month: String) -> Result<(), AppError> { schedule_service::clear_month_schedule(&state.db, &month).await }
